lib/index_documents.rb

```diff
@@ -0,0 +1,54 @@
+require_relative "../app"
+require 'gds_api/publishing_api_v2'
+
+class IndexDocuments
+  def process(message)
+    base_path = document_base_path(message)
+    topic_content_ids = topic_content_ids(message)
+
+    unless topic_content_ids.nil? # no topics field in links, don't update
+      topic_base_paths = topic_content_ids.map { |content_id| base_path(content_id) }
+      update_index(base_path, topic_base_paths)
+    end
+
+    message.ack
+  rescue StandardError => e
+    Airbrake.notify_or_ignore(
+      error: e,
+      payload: message.payload,
+      headers: message.headers,
+      delivery_info: message.delivery_info
+    )
+    message.retry
+  end
+
+  private
+
+  def document_base_path(message)
+    message.payload["base_path"]
+  end
+
+  def topic_content_ids(message)
+    message.payload["links"].try(:[], "topics")
+  end
+
+  def base_path(content_id)
+    GdsApi::PublishingApiV2.new(Plek.current.find('publishing-api')).get_content(content_id)["base_path"]
+  end
+
+  def update_index(base_path, topic_base_paths)
+    current_index.amend(base_path, { topics: topic_base_paths })
+  end
+
+  def current_index
+    @current_index ||= search_server.index(settings.default_index_name)
```

Comment:
I think the best approach would be to use `search_server.index_for_search(settings.search_config.content_index_names)` to first to a "get document", to check if the document exists at all and find out which index it's in.  Then use search_server.index(indexname) for the index returned in the data for that document to update the document in the right index.

This will also avoid errors being thrown by the call to `amend` if/when we get updates for documents which are excluded from the search index (for whatever reason).